### PR TITLE
Support `UPDATE ... FROM ( subquery )` in some dialects

### LINKED ISSUE #534: UPDATE ... FROM ( subquery ) only supported in PostgreSQL
Hi 👋 

The `UPDATE <table> FROM (SELECT ...)` statement is only supported with `PostgreSQL` dialect.

It should be also supported for other technologies. 
Here are a few examples:
- [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax#update_statement)
- [Snowflake](https://docs.snowflake.com/en/sql-reference/sql/update.html)
- [Redshift](https://docs.aws.amazon.com/redshift/latest/dg/c_Examples_of_UPDATE_statements.html)
- [SQL Server](https://docs.microsoft.com/fr-fr/sql/t-sql/queries/update-transact-sql?view=sql-server-ver16)

### PR DESCRIPTION
**Pull request by unvalley**
Closes #534 

Apply `UPDATE .. FROM (subquery)` for 
- BigQuery
- Redshift
- Snowflake
- SQL Server
- GenericDialect

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -5075,7 +5075,9 @@ impl<'a> Parser<'a> {
         let table = self.parse_table_and_joins()?;
         self.expect_keyword(Keyword::SET)?;
         let assignments = self.parse_comma_separated(Parser::parse_assignment)?;
-        let from = if self.parse_keyword(Keyword::FROM) && dialect_of!(self is PostgreSqlDialect) {
+        let from = if self.parse_keyword(Keyword::FROM)
+            && dialect_of!(self is PostgreSqlDialect | BigQueryDialect | SnowflakeDialect | RedshiftSqlDialect | MsSqlDialect)
```
Comment (AugustoFKL) [inline src/parser.rs:5079]: @unvalley please add the `GenericDialect` as well

File: tests/sqlparser_common.rs [hunk 2]
```diff
@@ -186,6 +186,93 @@ fn parse_update() {
     );
 }
 
+#[test]
+fn parse_update_set_from() {
+    let sql = "UPDATE t1 SET name = t2.name FROM (SELECT name, id FROM t1 GROUP BY id) AS t2 WHERE t1.id = t2.id";
+    let dialects = TestedDialects {
+        dialects: vec![
```
Comment (AugustoFKL) [inline tests/sqlparser_common.rs:193]: And add the generic dialect on tests as well please

**Review by AugustoFKL — CHANGES_REQUESTED** [review]

**Comment by unvalley** [comment]
@AugustoFKL Thank you for reviewing. I added GenericDialect.

File: tests/sqlparser_common.rs [hunk 3]
```diff
@@ -186,6 +186,94 @@ fn parse_update() {
...
+            Box::new(GenericDialect {}),
+            Box::new(PostgreSqlDialect {}),
+            Box::new(BigQueryDialect {}),
+            Box::new(RedshiftSqlDialect {}),
+            Box::new(MsSqlDialect {}),
```
Comment (AugustoFKL) [inline tests/sqlparser_common.rs:198]: @unvalley sorry I missed this, I think `Snowflake` is missing in tests, isn't it?

**Review by AugustoFKL — CHANGES_REQUESTED** [review]

File: tests/sqlparser_common.rs (hunk 3 above)
Comment (unvalley) [reply tests/sqlparser_common.rs:199]: It's my fault. Thank you for checking!
I added `Snowflake` in 413195b .

**Review by AugustoFKL — APPROVED** [review]
LGTM :)

Thanks for the quick response

**Comment by alamb** [comment]
I took the liberty of merging this PR to the master branch and resolving the conflicts (from #666) so I could include it in the release I am working on now

**Review by alamb — APPROVED** [review]

**Pull request merged**